README.md

```diff
@@ -3,8 +3,112 @@ A template for Rails 7, intended to help you quickly build apps for production w
 
 For the Rails 6 template please check the `rails-6` branch.
 
-### Usage
-run
+## Prerequisites:
+- [Node JS](https://nodejs.org/en)
+- A code editor, like [VS Code](https://code.visualstudio.com/)
+
+
+## Setup
+
+### MacOS
+1. Make sure you have [homebrew](https://docs.brew.sh/Installation) installed
+
+2. Install [rbenv](https://github.com/rbenv/rbenv) to install and manage ruby environments:
+```
+brew install rbenv ruby-install
+```
+
+3. Install the latest ruby version via rbenv and set it as the default version.
+The latest stable version is [3.2.2](https://www.ruby-lang.org/en/downloads/)
+
+```
+# install 3.2.2
+rbenv install 3.2.2
+# set the global version to 3.2.2
+rbenv global 3.2.2
+# set the local version to 3.2.2
+rbenv local 3.2.2
+# set the system version to 3.2.2
+rbenv shell 3.2.2
+```
+
+4. Restart the terminal
+```
+exec zsh
```

Comment:
this assumes having installed zsh too, which is not mentioned in the prerequisites (and probably shouldn't be)
probably best to reword into something like "If you are using Zsh you can run `exec zsh`"